### DISCUSSION TITLE
Compile and Test Moose Issue

### DISCUSSION THREAD

**Discussion by npaloskey** [idaholab/moose#20391] [Q&A Getting Started] [answered]
I keep getting: "TestHarness" requires python version 3.6 or greater, version 2.7 is being used." When attempting to do: 
./ run_tests -j4

When checking my version of of python in the shell using: python --version, it outputs Python 3.6.8.

Not sure what the issue here is. Any suggestions?

Note: I'm attempting to install moose on a job-scheduling environment and have been following the HPC Cluster installation guide as instructed by my research supervisor. I've had no issues up until this point.

Thanks!

**Comment by GiudGiud** [comment]
Hello

Did you alias python?
Is the python2 folder earlier than the python3 one in your PATH environment variable?

Guillaume

**Reply by npaloskey** [reply]
Hi, I didn't use an alias. Instead, my supervisor had told me to use "module load" in my .bash_profile. Below is what is currently in the 
.bash_profile:

# User specific environment and startup programs

PATH=$PATH:$HOME/bin

export PATH

module load gcc/8.3.1 mpich/3.2 cmake/3.18.4 python/3.6.8

source $HOME/.moose_profile

When returning to my home directory, I can run: "module list" and it outputs the following:

Currently Loaded Modules:
  1) gcc/8.3.1   2) mpich/3.2   3) cmake/3.18.4   4) python/3.6.8

I believe that the default python version for my system is 2.7 but I believe activating the above python module does change the version of python being used every time I open the shell.

Thanks for the quick reply!

**Reply by npaloskey** [reply]
Hey Giud,

I saw an earlier discussion in this thread addressing what sounds like the same problem I'm having currently. I changed the shebang in moose/test/run_tests to #!/usr/bin/env python3 and the it seems like most of the tests passed. Thanks!

**Reply by GiudGiud** [reply]
nice glad you got it going

**Comment by npaloskey** [comment, marked as answer]
Change the shebang in moose/test/run_tests to #!/usr/bin/env python3 and the tests should pass
